### ISSUE TITLE
emulator logs missing functionality of online console logs

### ISSUE THREAD

**Issue by ultraGentle** [firebase/firebase-tools#1859]
Hello, really enjoying firebase, but finding debugging a little bit painful.  To wit, 

It would be very helpful to at least have symbols like **LOG:**, **WARNING:**
### [REQUIRED] Environment info
**firebase-tools:** 7.9.0
**Platform:** macOS 10.13.6

### [REQUIRED] Test case
In index.js, create a  simple function with `console.log('log')`, `console.error('error')`

### [REQUIRED] Steps to reproduce
(1) Deploy function; check online logs. 
(2) Serve/emulate function locally; check terminal logs.  

### [REQUIRED] Expected behavior
Local logs should display clear differentiation between log types.  

### [REQUIRED] Actual behavior
`log` and `error` are not differentiated, making it difficult to visually scan what's going on.  

For some context, I'm a beginner when it comes to back-end debugging, but I'd like to believe that Firebase is at least partially targeted at users like me, who have a front-end background, and are looking for a friendly way to add serverless functionality.  

So, I'm used to looking at logs in the browser console, where `log`, `warn`, and `error` are presented visually differently, and can be filtered.  

This is an accessibility issue as well, since not everyone can easily visually scan a giant block of undifferentiated text (see #1839) -- the symbols and colors of browser devtools are a boon in that regard.  

So, to summarize, it would be extremely helpful to have better logging tools available locally, 
- symbols or text to differentiate log level
- separation of explicit user logging from auto-generated logs
- and at the bare minimum, to replicate the logging of the online console.  

I hope this is helpful feedback.  If I'm ignorant to a better way of debugging firebase functions, I would welcome suggestion.  And again, kudos on these tools.

PS: Just a final thought: perhaps when running `firebase serve`, the logs from index.js could be sent to the browser console?  I have no idea if that's possible, but if it were, it would certainly make life easier!

**Comment by ultraGentle** [comment]
Just to update this, when the `firebase serve` command is run, the output actually does have useful iconography: 

```
i  emulators: Starting emulators: functions, firestore, database, pubsub
⚠  Your requested "node" version "8" doesn't match your global version "12"
✔  functions: Emulator started at http://localhost:5001
i  firestore: Serving ALL traffic (including WebChannel) on http://localhost:8080
⚠  firestore: Support for WebChannel on a separate port (8081) is DEPRECATED and will go away soon. Please use port above instead.
i  firestore: Emulator logging to firestore-debug.log
✔  firestore: Emulator started at http://localhost:8080
```

So, symbols like that tied to `console.log`, `warn`, `error`, `info` would be great!

**Comment by samtstern** [comment]
@ultraGentle thanks for the feature request!

**Comment by ultraGentle** [comment]
Sorry, just realized this is a duplicate of #1800 !  I'll do any additional communicating there.

**Comment by christophe-g** [comment]
; ) Looks like a useful addition then !

**Comment by samtstern** [comment]
I updated the title of #1800 so we can close this one and continue discussing there, as @ultraGentle suggested.
